src/ImageSharp/Processing/Processors/Binarization/AdaptiveThresholdProcessor.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors
+{
+    /// <summary>
+    /// Performs Bradley Adaptive Threshold filter against an image
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format of the image</typeparam>
+    internal class AdaptiveThresholdProcessor<TPixel> : IImageProcessor<TPixel>
```

Comment:
Don't use `IImageProcessor<TPixel>` unless you know exactly what you're doing.

Reply:
Ok. I didn't wanna use ImageProcessor class because I don't quite understand how I can use pre, post methods in this context. I will read through that class file and try to use it as it was intended